### PR TITLE
Add new alpha: Greenblatts Magic Formula Alpha

### PR DESCRIPTION
**Pull request by HalldorAndersen**
#### Description
This alpha selects stocks according to Greenblatt's magic formula.

#### Motivation and Context
Create a new alpha for demonstration purposes.

#### How Has This Been Tested?
Tested in the cloud. Verified results and logic through logs and the alpha has been reviewed by the QuantConnect Team.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Algorithm.Python/Alphas/GreenblattMagicFormulaAlgorithm.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+        # Use MagicFormulaAlphaModel to establish insights
+        self.SetAlpha(MagicFormulaAlphaModel())
+
+        # Equally weigh securities in portfolio, based on insights
+        self.SetPortfolioConstruction(EqualWeightingPortfolioConstructionModel())
+        
+        # Select our default model types
+        self.SetPortfolioConstruction(EqualWeightingPortfolioConstructionModel())
+        self.SetExecution(ImmediateExecutionModel())
+        self.SetRiskManagement(NullRiskManagementModel())
```
Comment (AlexCatarino) [inline Algorithm.Python/Alphas/GreenblattMagicFormulaAlgorithm.py]: We could remove the default models (`ImmediateExecutionModel` and `NullRiskManagementModel`).

File: Algorithm.Python/Alphas/GreenblattMagicFormulaAlgorithm.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+from System import *
+from QuantConnect import *
+from QuantConnect.Orders import *
+from QuantConnect.Algorithm import QCAlgorithm
+from QuantConnect.Python import PythonQuandl
+
+from QuantConnect.Data.UniverseSelection import *
+from QuantConnect.Indicators import *
+from Selection.FundamentalUniverseSelectionModel import FundamentalUniverseSelectionModel
```
Comment (AlexCatarino) [inline Algorithm.Python/Alphas/GreenblattMagicFormulaAlgorithm.py]: Some of these imports are not being used in this algorithm:
- `from QuantConnect.Orders import *`
- `from QuantConnect.Algorithm import QCAlgorithm`
- `from QuantConnect.Python import PythonQuandl`
and add
- `from QuantConnect.Orders.Fees import ConstantFeeModel`

File: Algorithm.Python/Alphas/GreenblattMagicFormulaAlgorithm.py [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+#
+
+class MagicFormulaAlpha(QCAlgorithmFramework):
+    ''' Alpha Streams: Benchmark Alpha: Pick stocks according to Joel Greenblatt's Magic Formula'''
+
+    def Initialize(self):
+
+        self.SetStartDate(2018, 1, 1)
+        self.SetCash(100000)
+
```
Comment (AlexCatarino) [inline Algorithm.Python/Alphas/GreenblattMagicFormulaAlgorithm.py]: We are seeting the fees to zero.
```python
#Set zero transaction fees
self.SetSecurityInitializer(lambda security: security.SetFeeModel(ConstantFeeModel(0)))
```

File: Algorithm.Python/Alphas/GreenblattMagicFormulaAlgorithm.py [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+# First, each stock is ranked depending on the relative value of the ratio EV/EBITDA. For example, a stock
+# that has the lowest EV/EBITDA ratio in the security universe receives a score of one while a stock that has 
+# the tenth lowest EV/EBITDA score would be assigned 10 points.
+#
+# Then, each stock is ranked and given a score for the second valuation ratio, Return on Capital (ROC).
+# Similarly, a stock that has the highest ROC value in the universe gets one score point.
+# The stocks that receive the lowest combined score are chosen for insights.
+#
+# Source: Greenblatt, J. (2010) The Little Book That Beats the Market
+#
```
[the remark this answers is not in the document]
Comment (HalldorAndersen) [reply Algorithm.Python/Alphas/GreenblattMagicFormulaAlgorithm.py]: Correct.

**Pull request merged**